bigtable/client/internal/throw_delegate.cc

```diff
@@ -69,12 +65,14 @@ namespace internal {
 }
 
 [[noreturn]] void RaiseRpcError(grpc::Status const &status, char const *msg) {
-  // TODO(#119) - raise an exception that stores `status` as a value.
-  std::ostringstream os;
-  os << "unrecoverable gRPC error or too many gRPC errors in " << msg << ": "
-     << status.error_message() << " [" << status.error_code() << "] "
-     << status.error_details();
-  internal::RaiseRuntimeError(os.str());
+#ifdef GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  throw bigtable::GRpcError(msg, status);
+#else
+  bigtable::GRpcError ex(msg, status);
+  std::cerr << "Aborting because exceptions are disabled: " << ex.what()
+            << std::endl;
+  std::abort();
```

Comment:
1. Could you clarify the error handling strategy here from a users point of view?
2. In the exceptions case will the exception be raised all the way to the user, or will we catch it and do the appropriate thing in the bigtable client codebase?
3. What do we expect the user to do in the no exceptions case?

Reply:
The exceptions are raised all the way to the user.  If the user compiled without exceptions the error "handling" is we abort.  If the user is called "Google" there is an internal API to catch (most) errors without exceptions.  This pending PR has more context:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/pull/98/files